src/test/kotlin/com/mparticle/kits/GoogleAnalyticsFirebaseGA4KitTest.kt

```diff
@@ -121,6 +144,422 @@ class GoogleAnalyticsFirebaseGA4KitTest {
         TestCase.assertEquals(1, firebaseSdk.loggedEvents.size)
     }
 
+
+    @Test
+    fun onConsentStateUpdatedTest() {
+        val map = HashMap<String, String>()
+        map["defaultAdStorageConsentSDK"] = "Granted"
+        map["defaultAnalyticsStorageConsentSDK"] = "Granted"
+        map["consentMappingSDK"] =
+            "[{\\\"jsmap\\\":null,\\\"map\\\":\\\"Performance\\\",\\\"maptype\\\":\\\"ConsentPurposes\\\",\\\"value\\\":\\\"ad_user_data\\\"},{\\\"jsmap\\\":null,\\\"map\\\":\\\"Marketing\\\",\\\"maptype\\\":\\\"ConsentPurposes\\\",\\\"value\\\":\\\"ad_personalization\\\"},{\\\"jsmap\\\":null,\\\"map\\\":\\\"testconsent\\\",\\\"maptype\\\":\\\"ConsentPurposes\\\",\\\"value\\\":\\\"ad_storage\\\"}]"
+        map["defaultAdUserDataConsentSDK"] = "Denied"
+        map["defaultAdPersonalizationConsentSDK"] = "Denied"
+
+        kitInstance.configuration =
+            KitConfiguration.createKitConfiguration(JSONObject().put("as", map.toMutableMap()))
+
+
+        val locationCollectionConsent = GDPRConsent.builder(false)
+            .document("Test consent")
+            .location("17 Cherry Tree Lane")
+            .hardwareId("IDFA:a5d934n0-232f-4afc-2e9a-3832d95zc702")
+            .build()
+        val state = ConsentState.builder()
+            .addGDPRConsentState("Marketing", locationCollectionConsent)
+            .build()
+        filteredMParticleUser = FilteredMParticleUser.getInstance(user, kitInstance)
+
+        kitInstance.onConsentStateUpdated(state, state, filteredMParticleUser)
+
+        val expectedConsentValue =
+            firebaseSdk.getConsentState().getKeyByValue("AD_PERSONALIZATION").toString()
+        TestCase.assertEquals("DENIED", expectedConsentValue)
+
+    }
+
+    @Test
+    fun onConsentStateUpdatedTest_When_Marketing_true() {
+        val map = HashMap<String, String>()
+        map["defaultAdStorageConsentSDK"] = "Granted"
+        map["defaultAnalyticsStorageConsentSDK"] = "Granted"
+        map["consentMappingSDK"] =
+            "[{\\\"jsmap\\\":null,\\\"map\\\":\\\"Performance\\\",\\\"maptype\\\":\\\"ConsentPurposes\\\",\\\"value\\\":\\\"ad_user_data\\\"},{\\\"jsmap\\\":null,\\\"map\\\":\\\"Marketing\\\",\\\"maptype\\\":\\\"ConsentPurposes\\\",\\\"value\\\":\\\"ad_personalization\\\"},{\\\"jsmap\\\":null,\\\"map\\\":\\\"testconsent\\\",\\\"maptype\\\":\\\"ConsentPurposes\\\",\\\"value\\\":\\\"ad_storage\\\"}]"
+        map["defaultAdUserDataConsentSDK"] = "Denied"
+        map["defaultAdPersonalizationConsentSDK"] = "Denied"
+
+        kitInstance.configuration =
+            KitConfiguration.createKitConfiguration(JSONObject().put("as", map.toMutableMap()))
+
+        val locationCollectionConsent = GDPRConsent.builder(false)
+            .document("Test consent")
+            .location("17 Cherry Tree Lane")
+            .hardwareId("IDFA:a5d934n0-232f-4afc-2e9a-3832d95zc702")
+            .build()
+        val ccpaConsent =
```

Comment:
I don't think CCPA consent is relevant as a test as it does not pertain to the EU consent fix.